lnbits/core/services.py

```diff
@@ -396,11 +391,51 @@ class PaymentKwargs(TypedDict):
             checking_id="service_fee" + temp_id,
             payment_request=payment_request,
             payment_hash=invoice.payment_hash,
-            pending=False,
+            status=PaymentState.SUCCESS,
         )
     return invoice.payment_hash
 
 
+async def _create_external_payment(
+    temp_id: str,
+    amount_msat: MilliSatoshi,
+    conn: Optional[Connection],
+    **payment_kwargs,
+) -> Payment:
+    fee_reserve_total_msat = fee_reserve_total(amount_msat, internal=False)
+
+    old_payment = await get_standalone_payment(temp_id, conn=conn)
+    if old_payment:
+        if old_payment.pending:
+            raise PaymentError("Payment is still pending.", status="pending")
+        elif old_payment.success:
+            raise PaymentError("Payment already paid.", status="success")
+        elif old_payment.failed:
+            status = await old_payment.check_status()
```

Comment:
`check_status()` has this condition:

```py
        if self.failed:
            return PaymentFailedStatus()
```

will it ever make a call to the funding source in this scenario?

Reply:
where is that line of code? in my manual test it did work.

Reply:
yep, you were right, i removed this conditions, check should always go to fundingsource